src/test_over_serial/README.md

```diff
@@ -0,0 +1,160 @@
+
+### Input handler callback
+
+The `InputHandler` callback is used for application specific handling of received data.  A single parameter is passed to the callback, a pointer to the following data structure:
+```
+struct InputBuffer {
+  DataPtr data;   // input buffer pointer
+  size_t length;  // input buffer length
+  size_t offset;  // offset from start of input
+  size_t total;   // total data that will be transferred
+};
+```
+All members of this data structure will have values in units of the `data-type`.  For example, 16-bit mono audio data will consist of 2 bytes per audio `sample`.  The units of all values in the data structure would therefore be a count of `sample(s)`, not bytes.
+
+The input buffer data will have been base-64 decoded prior to the `InputHandler` callback being executed.
+
+The `InputHandler` callback must return `true` to allow data transfer to continue.  Returning `false` will abort the data transfer and a `!FAIL DATA...` response will be sent to the serial interface.
+
+## Testing with test_over_serial.py script
+
+The [test_over_serial.py](/scripts/test_over_serial.py) script is provided for automated testing of the supported Arduino example applications.
+
+Only the `--example` option is required, which specifies which example application to test.  The `--verbose` option controls the amount of output seen.  The `--verbose` values can be:
+- `all` See all output including any output sent by the Arduino application
+- `test` Restrict output to just the test results
+
+Example command line (must be run from the top of the tflite-micro-arduino-examples repository directory):
+```
+tflite-micro-arduino-examples$ python3 scripts/test_over_serial.py --example person_detection --verbose test
+10:21:03.746946  Test start:
+10:21:04.016295  Test #1 for label <person> file <person.bmp>
+10:21:05.280497  Test #1 FAILED
+10:21:05.280648  Waiting 4.0s before next test
+10:21:09.288489  Test #2 for label <no person> file <no_person.bmp>
+10:21:10.550045  Test #2 PASSED
+10:21:10.550116  Waiting 4.0s before next test
+10:21:14.556783  Test #3 for label <person> file <person.bmp>
+10:21:16.328190  Test #3 PASSED
+10:21:16.328250  Waiting 4.0s before next test
+10:21:20.334084  Test #4 for label <no person> file <no_person.bmp>
+10:21:22.115927  Test #4 PASSED
+10:21:22.222481  Test end: total 4 completed 4 passed 3 failed 1
+```
+### Linux and `ModemManager`
+
+If `!TEST` command timeouts are seen when running the test script, it may be that `ModemManager` is interfering with the serial interface.  `ModemManager` will need to be disabled or removed.
+
+To disable (Ubuntu Linux):
+```
+systemctl disable ModemManager.service
+systemctl stop ModemManager.service
+```
+
+To remove (Ubuntu Linux):
+```
+sudo apt-get purge modemmanager
+```
+
+### Example test configuration data
+
+The test script relies on configuration data in a JSON compatible form. An example configuration:
+```
+{
```

Comment:
Do you think it would make sense to put test configurations alongside sample images / audio clips in examples/<example>/data directory? I would personally find it easier to work from an existing json file rather than copying from the README.

Reply:
Will be added in the next PR (micro_speech).